src/include/storage/sql_table.h

```diff
@@ -174,15 +319,19 @@ class SqlTable {
   BlockStore *const block_store_;
   const catalog::table_oid_t oid_;
 
-  // Eventually we'll support adding more tables when schema changes. For now we'll always access the one DataTable.
-  DataTableVersion table_;
+  common::ConcurrentMap<layout_version_t, DataTableVersion> tables_;
+  // NOTE: This map only keeps track of the default values specified at column creation
+  // For columns which don't have default value or added later, just set to null
+  // Populating default values into the ProjectedRow inserted later is taken care of by the execution engine
+  DefaultValueMap default_value_map_;
```

Comment:
Users can drop columns. So some default values may be no longer needed in the future. You may want to have some way to shrink the `DefaultValueMap`.